### PR TITLE
fix: move wrapping in response from daos to handlers

### PR DESCRIPTION
**Pull request by marybrown123**

### CODE REVIEW

File: src/recipe/queries/handlers/findAllRecipes.handler.ts [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+  ) {}
 
   async execute(query: FindAllRecipesQuery) {
     const { queryData } = query;
-    return this.recipeDAO.findAllRecipes(queryData);
+    const recipesFromDb = await this.recipeDAO.findAllRecipes(queryData);
+
+    return Promise.all(
+      recipesFromDb.map(async (recipe) => {
+        const file = await this.fileService.findFileById(recipe.fileId);
```
Comment (wojdanowski) [inline src/recipe/queries/handlers/findAllRecipes.handler.ts:24]: what if there is no file on the recipe? are there any test cases covering that?

File: src/recipe/recipe.dao.ts [hunk 2]
```diff
@@ -93,13 +100,14 @@ export class RecipeDAO {
...
-
-    return new RecipeResponse(recipeFromDb, fileUrl);
+    return recipeFromDb;
   }
 
-  async findAllRecipes(query: FindAllRecipesDTO): Promise<RecipeResponse[]> {
+  async findAllRecipes(query: FindAllRecipesDTO): Promise<
+    (Recipe & { preparing: RecipePreparationSteps[] } & {
+      ingredients: RecipeIngredients[];
+    })[]
```
Comment (wojdanowski) [inline src/recipe/recipe.dao.ts:109]: please consider moving those return types to some kind of interface or type, DRY

File: src/recipe/test/unit/recipe.service.spec.ts [hunk 3]
```diff
@@ -160,7 +160,7 @@ describe('Recipe Service', () => {
       limit: 2,
     });
 
-    expect(queryBusExecuteFindMany).toBeCalledTimes(1);
+    expect(queryBusExecuteFindMany).toBeCalledTimes(2);
```
Comment (wojdanowski) [inline src/recipe/test/unit/recipe.service.spec.ts:163]: name of that spy indicates that you spy on a particular query (queryBusExecuteFindMany) from queryBus, in fact, you spay on 'execute'. Please consider renaming this const

File: src/recipe/queries/handlers/findAllRecipes.handler.ts (hunk 1 above)
Comment (marybrown123) [reply src/recipe/queries/handlers/findAllRecipes.handler.ts:24]: there. must be file on the recipe

**Pull request merged**